### ISSUE TITLE
investigate tooling to check for a11y issues

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-next#466]
Investigate what tools are available that can help us flag accessibility issues that may be present in a component that is being implemented for PatternFly 4. Provide any information that would be helpful in deciding between tools, if there is more than one tool that would work.

Done Criteria:
- Select and agree on tool for UXD to use to verify a11y violations
- PR with a base solution posted against https://github.com/patternfly/patternfly-next/
- Issues opened for remaining work items

Requirements:
- Tooling should be usable without needing to write code or run scripts
- Tooling should be scriptable and able to run as part of CI jobs
- Tooling should be JavaScript framework agnostic
- Tooling should be able to analyze static html pages
- Tooling should allow for report formats that are easy to understand
- Tooling should allow for analyzing pages against WCAG2.1 AA

Nice to have:
- Tooling should be able to analyze pages with dynamic content
- Tooling should allow for analyzing pages against multiple standards like Section508, WCAG2.0 AA, WCAG2.0 AAA, etc.
- Tooling is ideally testing framework agnostic, or at least compatible with leading testing frameworks
- Tooling should be able to optionally fail a build when it finds a number of violations
- Tooling can include/exclude target sections of DOM
- Tooling can target specific rules like color contrast, accessible names, etc.

**Comment by seanforyou23** [comment]
And the winner (my suggestion) is: aXe: The Accessibility Engine

Through my investigation, aXe became the obvious choice for us moving forward. For many reasons, but among the top were:

1. aXe browser extension that non-technical users will enjoy - https://chrome.google.com/webstore/detail/axe/lhdoppojpmngadmnindnejefpokejbdd and https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/
2. Can be used with selenium, mocha, phantomJS, etc. for when you need fine grain control, scripting, custom reporting, etc.
3. aXe CLI makes overnight integration possible
4. Recommended in https://www.w3.org/WAI/ER/tools/ and  https://reactjs.org/docs/accessibility.html
5. Allows for targeting specific standards
6. Allows for excluding certain rules
7. Allows for specifying certain sections of DOM to analyze
8. Allows for spot testing pages
9. Does not require already having an automated testing workflow set up to use
10. aXe is open source, and actively involved in the community - hosting webcasts, posting blogs and tutorials, and responding quickly to github issues

Screenshots below to illustrate what working with this tool can look like;

<img width="1058" alt="screen shot 2018-08-01 at 5 03 53 pm" src="https://user-images.githubusercontent.com/5942899/43548855-04550272-95ad-11e8-8bc9-cc9aa6bfcae1.png">

<img width="930" alt="screen shot 2018-08-01 at 4 50 36 pm" src="https://user-images.githubusercontent.com/5942899/43548868-0d445de2-95ad-11e8-8d72-5a6c85fcddaa.png">

<img width="913" alt="screen shot 2018-08-01 at 1 53 23 pm" src="https://user-images.githubusercontent.com/5942899/43548894-1ec2a86c-95ad-11e8-8816-32f32384be84.png">
